### PR TITLE
Adds InitialMarginRequirement and MaintenanceMarginRequirement Properties to FutureMarginModel

### LINKED ISSUE #4023: Add Initial and Maintenance Margin Properties to FuturesMarginModel
#### Expected Behavior
Currently impossible to calculate number of contracts you can afford with cash.

#### Actual Behavior
Ability to use the initial margin requirement to calculate the number of contracts you can purchase.

#### Potential Solution
 - Add symbol/security to the model; 
 - Create 2 new properties InitialMargin and MainteanceMargin decimals
 - Expose the values at the time of the property get; 

#### Checklist 
- [x] I have completely filled out this template
- [x] I have confirmed that this issue exists on the current `master` branch
- [x] I have confirmed that this is not a duplicate issue by searching [issues](https://github.com/QuantConnect/Lean/issues)

### PR DESCRIPTION
**Pull request by AlexCatarino**
#### Description
These properties expose the current margin requirements that can be used to compute the number of contracts manually.

#### Related Issue
Closes #4023 

#### Motivation and Context
Provide access to margin requirements of futures that change with time.

#### Requires Documentation Change
Yes. It would be interesting to show that these properties exist.

#### How Has This Been Tested?
Regression algorithm.

#### Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Refactor (non-breaking change which improves implementation)
- [ ] Performance (non-breaking change which improves performance. Please add associated performance test and results)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Non-functional change (xml comments/documentation/etc)

#### Checklist:
- [x] My code follows the code style of this project.
- [x] I have read the **CONTRIBUTING** [document](https://github.com/QuantConnect/Lean/blob/master/CONTRIBUTING.md).
- [x] All new and existing tests passed.
- [x] My branch follows the naming convention `bug-<issue#>-<description>` or `feature-<issue#>-<description>`

### CODE REVIEW

File: Algorithm.CSharp/BasicTemplateFuturesAlgorithm.cs [hunk 1]
```diff
@@ -81,6 +82,11 @@ select futuresContract
                     // if found, trade it
                     if (contract != null)
                     {
+                        // Get the margin requirements
+                        var model = Securities[contract.Symbol].BuyingPowerModel as FutureMarginModel;
+                        var initialOvernight = model?.InitialMarginRequirement;
+                        var maintenanceOvernight = model?.MaintenanceMarginRequirement;
```
Comment (gsalaz98) [inline Algorithm.CSharp/BasicTemplateFuturesAlgorithm.cs]: Nit: would be nice to have some assertions that these are (or aren't) null

File: Algorithm.Python/BasicTemplateFuturesAlgorithm.py [hunk 2]
```diff
@@ -57,6 +57,12 @@ def OnData(self,slice):
                 # if there is any contract, trade the front contract
                 if len(contracts) == 0: continue
                 front = sorted(contracts, key = lambda x: x.Expiry, reverse=True)[0]
+
+                # Get the margin requirements
+                model = self.Securities[front.Symbol].BuyingPowerModel
+                initialMarginRequirement = model.InitialMarginRequirement
+                maintenanceMarginRequirement = model.MaintenanceMarginRequirement
```
Comment (gsalaz98) [inline Algorithm.Python/BasicTemplateFuturesAlgorithm.py]: Nit: same as BasicTemplateFuturesAlgorithm.cs comments

File: Common/Brokerages/DefaultBrokerageModel.cs [hunk 3]
```diff
@@ -300,7 +300,7 @@ public virtual IBuyingPowerModel GetBuyingPowerModel(Security security)
                     model = new OptionMarginModel(RequiredFreeBuyingPowerPercent);
                     break;
                 case SecurityType.Future:
-                    model = new FutureMarginModel(RequiredFreeBuyingPowerPercent);
+                    model = new FutureMarginModel(RequiredFreeBuyingPowerPercent, security);
```
Comment (gsalaz98) [inline Common/Brokerages/DefaultBrokerageModel.cs]: Are we missing a security argument in `AlphaStreamsBrokerageModel`? It looks similar to the `DefaultBrokerageModel`

Comment (gsalaz98) [reply Common/Brokerages/DefaultBrokerageModel.cs]: Also, in the `Future` security class, should we be providing it to the multiple `FutureMarginModel` that we construct?

Comment (AlexCatarino) [reply Common/Brokerages/DefaultBrokerageModel.cs]: The `BuyingPowerModel` that is set in the `Future` constructor is overridden by the brokerage model.

**Pull request merged**